### PR TITLE
Create user profile endpoint (old)

### PR DESCRIPTION
**Pull request by seanim0920**

### CODE REVIEW

**Review by suryatiru — DISMISSED** [review]

File: APILambda/auth.ts [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
-  message: "Unauthorized"
+  error: "Unauthorized"
 }
 
 /**
  * Adds AWS cognito token verification to an app.
  */
-export const addCognitoTokenVerification = (app: Application) => {
+export const addCognitoTokenVerification = (app: Application, env: ServerEnvironment) => {
   // TODO: - Verify JWT properly
```
Comment (seanim0920) [inline APILambda/auth.ts:24]: Try jsonwebtoken

File: APILambda/auth.ts [hunk 2]
```diff
@@ -1,29 +1,50 @@
...
+    // TODO: perform JWT verification if envType !== dev
 
-    const token = event.headers.Authorization.split(" ")[1]
+    const token = auth.split(" ")[1] // TODO: ensure correct format of auth header
 
     const claims = JSON.parse(
       Buffer.from(token.split(".")[1], "base64").toString()
     )
 
     res.locals.selfId = claims!.sub
```
Comment (seanim0920) [inline APILambda/auth.ts]: Use zod to parse claims

File: APILambda/auth.ts [hunk 3]
```diff
@@ -1,29 +1,50 @@
...
-  app.use((_, res, next) => {
-    const { event } = getCurrentInvoke()
-    if (!event?.headers?.Authorization) {
+  app.use((req, res, next) => {
+    let auth = req.headers.Authorization
+    if (env.environment === "dev") {
+      auth = req.headers.authorization
+    }
+
+    if (!auth || Array.isArray(auth)) {
```
Comment (seanim0920) [inline APILambda/auth.ts:31]: Add unit tests for edge cases

File: APILambda/user/createUserProfile.test.ts [hunk 4]
```diff
@@ -1,47 +1,57 @@
...
 
-  it("should 201 when creating a user with a unique id and handle", async () => {
-    const resp = await callPostUser(testUserIdentifier, testUsers[0])
+  it("should 401 when user is missing 'name' attribute", async () => {
+    const resp = await callPostUser(generateMockToken({ ...mockClaims, name: "" }))
 
-    expect(resp.status).toEqual(201)
-    expect(resp.body).toMatchObject({ id: testUserIdentifier })
+    expect(resp.status).toEqual(401)
+    expect(resp.body).toMatchObject({ error: "Unauthorized" })
```
Comment (seanim0920) [inline APILambda/user/createUserProfile.test.ts:20]: Return user id and handle

File: APILambda/user/userHandle.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * ```ts
+ * isValidUserHandle("@") // 🔴 false, has no characters after @
+ * isValidUserHandle("dkjbd") // 🔴 false, doesn't start with @
+ * isValidUserHandle("@+++test+++") // 🔴 false, not alpha numeric
+ * isValidUserHandle("@aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa") // 🔴 false, too long
+ * isValidUserHandle("@why_people") // ✅ true
+ * ```
+ */
+export const isValidUserHandle = (handle: string) => {
```
Comment (seanim0920) [inline APILambda/user/userHandle.ts:37]: not necessary

**Review by suryatiru — APPROVED** [review]
